File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    q1 = deque(progresses)
    q2 = deque(speeds)
    while True:
        if not q1: break
        
        tmp = []
        for i in range(len(q1)):
            q1[i] += q2[i]
        while q1 and q1[0] >= 100:
            tmp.append(q1.popleft())
            q2.popleft()
        if tmp:
            answer.append(len(tmp))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> ans = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int tmp = (100 - progresses[i]) % speeds[i];
            if (tmp == 0) {
                q.add((100 - progresses[i]) / speeds[i]);
            } else {
                q.add((100 - progresses[i]) / speeds[i] + 1);
            }
        }
        
        int previous = q.poll();
        int cnt = 1;
        while(!q.isEmpty()) {
            if (q.peek() <= previous) {
                q.poll();
                cnt++;
            } else {
                ans.add(cnt);
                previous = q.poll();
                cnt = 1;
            }
        }
        ans.add(cnt);
        
        return ans.stream().mapToInt(e->e).toArray();
    }
}